Google.PowerShell/BigQuery/BqTable.cs

```diff
@@ -330,4 +330,174 @@ public TablesResource.InsertRequest makeInsertReq()
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset to search. This dataset should contain the table you wish to remove.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.ByValue)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Dataset that you would like to search. This field takes Dataset or DatasetRefrence objects.
+        /// This dataset should contain the table you wish to remove.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        public DatasetReference Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Table object that will be sent to the server to be removed. 
+        /// Also takes TableReference and TableList.TablesData objects.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(TableList.TablesData),
+            Property = nameof(TableList.TablesData.TableReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Apis.Bigquery.v2.Data.Table.TableReference))]
+        public TableReference InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Forces deletion of non-empty tables and the data contained in them.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByValue:
+                    // No processing needed for ByValue parameter sets.
+                    break;
+                case ParameterSetNames.ByObject:
+                    Project = InputObject.ProjectId;
+                    DatasetId = InputObject.DatasetId;
+                    Table = InputObject.TableId;
+                    break;
+                case ParameterSetNames.ByDatasetObject:
+                    Project = Dataset.ProjectId;
+                    DatasetId = Dataset.DatasetId;
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            TablesResource.DeleteRequest request = Service.Tables.Delete(Project, DatasetId, Table);
+            String response = "Table Removal Stopped.";
```

Comment:
We don't want to pollute the output pipeline with an information string. I would refactor this code with a new function contains all of the should delete logic and returns a boolean. Then, if it returns true, execute the delete request. I'm not sure this cmdlet should write anything to the pipeline. The response says it is a string, but I suspect it is always an empty one.